compiler/rustc_middle/src/middle/exported_symbols.rs

```diff
@@ -36,6 +36,7 @@ pub struct SymbolExportInfo {
     pub level: SymbolExportLevel,
     pub kind: SymbolExportKind,
     pub used: bool,
+    pub used_compiler: bool,
```

Comment:
Please add a doc comment explaining why there are two `used` fields and what they each do.